### ISSUE TITLE
Inconsistent handling of typed literals in sparql-update

### ISSUE THREAD

**Issue by mrkvon** [nodeSolidServer/node-solid-server#1723]
### TL;DR

Subsequent `INSERT DATA` and `DELETE DATA` of the same triple with object `"1234.56789e0"^^<http://www.w3.org/2001/XMLSchema#double>` leads to response status `409 Conflict`.

### actual behavior

I send a patch request on an empty or non-existent resource (newlines added for easier readability):

```curl
curl 'https://testuser.solidcommunity.net/path/to/resource' \
-X PATCH \
[...]
-H 'content-type: application/sparql-update' \
-H 'authorization: DPoP [...]' \
-H 'dpop: [...]' \
[...]
--data-raw $'INSERT DATA {
    <https://testuser.solidcommunity.net/path/to/resource#location>
        <http://www.w3.org/1999/02/22-rdf-syntax-ns#type>
            <http://www.w3.org/2003/01/geo/wgs84_pos#Point> .

    <https://testuser.solidcommunity.net/path/to/resource#location>
        <http://www.w3.org/2003/01/geo/wgs84_pos#lat>
            "60.235039190740146"^^<http://www.w3.org/2001/XMLSchema#double> .

    <https://testuser.solidcommunity.net/path/to/resource#location>
        <http://www.w3.org/2003/01/geo/wgs84_pos#long>
            "24.941368103027344"^^<http://www.w3.org/2001/XMLSchema#double> .
}'
```

And a resource with the following content is produced:

```turtle
@prefix : <#>.
@prefix geo: <http://www.w3.org/2003/01/geo/wgs84_pos#>.

:location
a geo:Point; geo:lat 60.235039190740146e0; geo:long 24.941368103027344e0 .
```

_Please note that typed literals were replaced with numbers._

Then i send another request (`INSERT DATA` changed to `DELETE DATA`):

```curl
curl 'https://testuser.solidcommunity.net/path/to/resource' \
-X PATCH \
[...]
-H 'content-type: application/sparql-update' \
-H 'authorization: DPoP [...]' \
-H 'dpop: [...]' \
[...]
--data-raw $'DELETE DATA {
    <https://testuser.solidcommunity.net/path/to/resource#location>
        <http://www.w3.org/1999/02/22-rdf-syntax-ns#type>
            <http://www.w3.org/2003/01/geo/wgs84_pos#Point> .

    <https://testuser.solidcommunity.net/path/to/resource#location>
        <http://www.w3.org/2003/01/geo/wgs84_pos#lat>
            "60.235039190740146"^^<http://www.w3.org/2001/XMLSchema#double> .

    <https://testuser.solidcommunity.net/path/to/resource#location>
        <http://www.w3.org/2003/01/geo/wgs84_pos#long>
            "24.941368103027344"^^<http://www.w3.org/2001/XMLSchema#double> .
}'
```

and the request fails with status code `409 Conflict` and body

```
The patch could not be applied. Could not find to delete: <https://testuser.solidcommunity.net/path/to/resource#location> <http://www.w3.org/2003/01/geo/wgs84_pos#lat> "60.235039190740146e0"^^<http://www.w3.org/2001/XMLSchema#double> .
 or <https://testuser.solidcommunity.net/path/to/resource#location> <http://www.w3.org/2003/01/geo/wgs84_pos#long> "24.941368103027344e0"^^<http://www.w3.org/2001/XMLSchema#double> .
```

When typed literals are changed to numbers, response is `200`.

### expected behavior

Sending subsequent `INSERT DATA {}` and `DELETE DATA {}` with identical body should succeed.

Either the server should save the typed literal as typed literal, or it should treat typed literal "double" and number as identical for the purposes of `DELETE DATA`. Intuitively, the former feels more correct.

### NSS version

Account on https://solidcommunity.net (2023-03-17)

```
Name
    solidcommunity.net
Description
    An experimental solid server run by the community
Details
    Running on [Node Solid Server 5.7.6](https://github.com/solid/node-solid-server/releases/tag/v5.7.6)
```

**Comment by mrkvon** [comment]
On the second look, i noticed that number in `DELETE DATA` request contained additional `e0`, e.g. `"1234.56789e0"^^<http://www.w3.org/2001/XMLSchema#double>`. When this `e0` was removed, the request passed as expected and the data were correctly deleted.

So the NSS behaves consistently after all. Now, all that remains is the surprise that `"123.456"^^xsd:double` becomes `123.456e0`.

So the remaining questions are:
- is `123.456e0` equivalent to `"123.456"^^xsd:double`?
- is `123.456e0` equivalent to `"123.456e0"^^xsd:double`? If not, why not?

I resolved it by using `xsd:decimal`, which was good enough. NSS then saved the number with the type tag, as i would initially expect. And Comunica doesn't go funny... :)

**Comment by mrkvon** [comment]
I guess if someone can verify that `123.456e0` is equivalent to `"123.456"^^xsd:double`, and not equivalent to `"123.456e0"^^xsd:double`, then this can be closed as non-issue.

**Comment by bourgeoa** [comment]
@angelo-v I think you worked with the E representation in rdflib.
